src/main/java/de/tum/in/www1/artemis/service/quiz/QuizResultService.java

```diff
@@ -0,0 +1,190 @@
+                    }
+
+                    participation.setInitializationState(InitializationState.FINISHED);
+
+                    boolean resultExisting = false;
+                    // create new result if none is existing
+                    Result result;
+                    if (participation.getResults().isEmpty()) {
+                        result = new Result().participation(participation);
+                    }
+                    else {
+                        resultExisting = true;
+                        result = participation.getResults().iterator().next();
+                    }
+                    // Only create Results once after the first evaluation
+                    if (!resultExisting) {
+                        // delete result from quizSubmission, to be able to set a new one
+                        if (quizSubmission.getLatestResult() != null) {
+                            resultService.deleteResult(quizSubmission.getLatestResult(), true);
+                        }
+                        result.setRated(true);
+                        result.setAssessmentType(AssessmentType.AUTOMATIC);
+                        result.setCompletionDate(ZonedDateTime.now());
+
+                        // set submission to calculate scores
+                        result.setSubmission(quizSubmission);
+                        // calculate scores and update result and submission accordingly
+                        quizSubmission.calculateAndUpdateScores(quizExercise.getQuizQuestions());
+                        result.evaluateQuizSubmission(quizExercise);
+                        // remove submission to follow save order for ordered collections
+                        result.setSubmission(null);
+
+                        // NOTE: we save participation, submission and result here individually so that one exception (e.g. duplicated key) cannot destroy multiple student answers
+                        submissionRepository.save(quizSubmission);
+                        result = resultRepository.save(result);
+
+                        // add result to participation
+                        participation.addResult(result);
+                        studentParticipationRepository.save(participation);
+
+                        // add result to submission
+                        result.setSubmission(quizSubmission);
+                        quizSubmission.addResult(result);
+                        submissionRepository.save(quizSubmission);
+
+                        // Add result so that it can be returned (and processed later)
+                        createdResults.add(result);
+                    }
+                }
+                catch (Exception e) {
+                    log.error("Exception in evaluateExamQuizExercise() for user {} in quiz {}: {}", participation.getParticipantIdentifier(), quizExercise.getId(), e.getMessage(),
+                            e);
```

Comment:
Maybe also notify the instructor? Otherwise the instructor will not notice that one calculation of 2000 failed

Reply:
This was already the previous behavior and was just copied to another place.
I talked with Stephan and we decided that this would not be part of this PR, as a good solution would need larger refactoring.